src/trpc/routers/template-field-router/procedures/add-fields.ts

```diff
@@ -1,152 +1,108 @@
-import { sendMail } from "@/server/mailer";
-import { SignJWT, jwtVerify } from "jose";
-import { render } from "jsx-email";
-import { z } from "zod";
-
-interface SendEmailOptions {
-  email: string;
-  token: string;
-}
-
-const emailTokenPayloadSchema = z.object({
-  id: z.string(),
-  rec: z.string(),
-});
-
-type TEmailTokenPayloadSchema = z.infer<typeof emailTokenPayloadSchema>;
-
-interface EncodeEmailTokenOption {
-  templateId: string;
-  recipientId: string;
-}
-
-const secret = new TextEncoder().encode(env.NEXTAUTH_SECRET);
-
-export function EncodeEmailToken({
-  recipientId,
-  templateId,
-}: EncodeEmailTokenOption) {
-  const encodeToken: TEmailTokenPayloadSchema = {
-    rec: recipientId,
-    id: templateId,
-  };
-
-  return new SignJWT(encodeToken)
-    .setProtectedHeader({ alg: "HS256" })
-    .sign(secret);
-}
-
-export async function DecodeEmailToken(jwt: string) {
-  const { payload } = await jwtVerify(jwt, secret);
-
-  return emailTokenPayloadSchema.parse(payload);
-}
-
-export async function SendEsignEmail({ email, token }: SendEmailOptions) {
-  const baseUrl = env.NEXTAUTH_URL;
-  const html = await render(
-    EsignEmail({
-      signingLink: `${baseUrl}/documents/esign/${token}`,
-    }),
-  );
-  await sendMail({
-    to: email,
-    subject: "esign Link",
-    html,
-  });
-}
-
 export const addFieldProcedure = withAuth
   .input(ZodAddFieldMutationSchema)
   .mutation(async ({ ctx, input }) => {
-    const mails: Promise<void>[] = [];
-
-    await ctx.db.$transaction(async (tx) => {
-      const { companyId } = await checkMembership({
-        tx,
-        session: ctx.session,
-      });
-
-      const template = await tx.template.findFirstOrThrow({
-        where: {
-          publicId: input.templatePublicId,
-          companyId,
-          status: "DRAFT",
-        },
-        select: {
-          id: true,
-          orderedDelivery: true,
-        },
-      });
-      await tx.templateField.deleteMany({
-        where: {
-          templateId: template.id,
-        },
-      });
-
-      const recipientList = await tx.esignRecipient.findMany({
-        where: {
-          templateId: template.id,
-        },
-        select: {
-          id: true,
-          email: true,
-        },
-      });
-
-      const fieldsList = [];
-
-      for (let index = 0; index < input.data.length; index++) {
-        const field = input.data[index];
-
-        if (field) {
-          fieldsList.push({ ...field, templateId: template.id });
-        }
+    try {
+      const user = ctx.session.user;
+      const companyLogo = input.emailPayload.company.logo;
+
+      if (input.completedOn) {
+        return {
+          success: false,
+          message: "E-signing has already completed among all parties.",
+        };
+      }
+      if (!user.email || !user.name || !companyLogo) {
+        return {
+          success: false,
+          message: "Email requires sender name , email and company logo.",
+        };
```

Comment:
Let's not make company logo a requirement to eSign. We have a placeholder we can use if company logo is not available.

`/public/placeholders/company.svg`